display_list/display_list_image_filter_unittests.cc

```diff
@@ -759,6 +771,63 @@ TEST(DisplayListImageFilter, UnknownContents) {
   ASSERT_EQ(filter.skia_object().get(), sk_filter.get());
 }
 
+TEST(DisplayListImageFilter, LocalImageFilterBounds) {
+  auto inputBounds = SkIRect::MakeLTRB(20, 20, 80, 80);
+  auto filter_matrix = SkMatrix::MakeAll(2.0, 0.0, 10,  //
+                                         0.5, 3.0, 15,  //
+                                         0.0, 0.0, 1);
+
+  auto sk_filter1 =
+      SkImageFilters::Blur(5.0, 6.0, SkTileMode::kRepeat, nullptr);
+  auto sk_filter2 = SkImageFilters::ColorFilter(
+      SkColorFilters::Blend(SK_ColorRED, SkBlendMode::kSrcOver), nullptr);
+  auto sk_filter3 = SkImageFilters::Dilate(5.0, 10.0, nullptr);
+  auto sk_filter4 = SkImageFilters::MatrixTransform(
+      filter_matrix, ToSk(DlImageSampling::kLinear), nullptr);
+  auto sk_filter5 = SkImageFilters::Compose(sk_filter1, sk_filter2);
+  std::vector<sk_sp<SkImageFilter>> sk_filters{
+      sk_filter1, sk_filter2, sk_filter3, sk_filter4, sk_filter5};
+
+  DlBlendColorFilter dl_color_filter(DlColor::kRed(), DlBlendMode::kSrcOver);
+  auto dl_filter1 =
+      std::make_shared<DlBlurImageFilter>(5.0, 6.0, DlTileMode::kRepeat);
+  auto dl_filter2 =
+      std::make_shared<DlColorFilterImageFilter>(dl_color_filter.shared());
+  auto dl_filter3 = std::make_shared<DlDilateImageFilter>(5, 10);
+  auto dl_filter4 = std::make_shared<DlMatrixImageFilter>(
+      filter_matrix, DlImageSampling::kLinear);
+  auto dl_filter5 = std::make_shared<DlComposeImageFilter>(
+      dl_filter1->shared(), dl_filter2->shared());
+  std::vector<std::shared_ptr<DlImageFilter>> dl_filters{
+      dl_filter1, dl_filter2, dl_filter3, dl_filter4, dl_filter5};
+
+  auto translate = SkMatrix::Translate(10.0, 10.0);
+  auto scale_translate = SkMatrix::Scale(2.0, 2.0).preTranslate(10.0, 10.0);
+  auto rotate_translate = SkMatrix::RotateDeg(45).preTranslate(5.0, 5.0);
+  auto persp = SkMatrix::I();
+  persp.setPerspY(10);
+  std::vector<SkMatrix> matrixs = {translate, scale_translate, rotate_translate,
+                                   persp};
+
+  for (unsigned i = 0; i < sk_filters.size(); i++) {
+    for (unsigned j = 0; j < matrixs.size(); j++) {
+      auto& m = matrixs[j];
+      auto sk_local_filter = sk_filters[i]->makeWithLocalMatrix(m);
+      auto dl_local_filter = dl_filters[i]->makeWithLocalMatrix(m);
+      SkIRect sk_rect, dl_rect;
+      if (sk_local_filter) {
+        sk_rect = sk_local_filter->filterBounds(
+            inputBounds, SkMatrix::I(),
```

Comment:
Don't use SkMatrix::I here. Use "for matrix in matrices" and test under all types of ctms.